what4/src/What4/Utils/Process.hs

```diff
@@ -68,13 +70,25 @@ withProcessHandles path args mcwd action = do
           (Just in_h, Just out_h, Just err_h, ph) -> return (in_h,out_h,err_h,ph)
           _ -> fail "Internal error in withProcessHandles: Failed to create handle."
   let cleanup (in_h,out_h,err_h,ph) = do
-        catch (do hClose in_h
+        catchJust filterAsync
+              (do hClose in_h
                   hClose out_h
-                  hClose err_h
-                  void $ waitForProcess ph)
-              (\(_ :: SomeException) -> return ())
+                  hClose err_h)
+              (\(ex :: SomeException) -> hPutStrLn stderr $ displayException ex)
+        void $ waitForProcess ph
+
   let onError (_,_,_,ph) = do
         -- Interrupt process; suppress any exceptions that occur.
-        catch (terminateProcess ph) (\(_ :: SomeException) -> return ())
+        catchJust filterAsync (terminateProcess ph) (\(ex :: SomeException) ->
```

Comment:
Just so I understand here: we are propagating Async exceptions from the call to `terminateProcess`, but just swallowing failures.  Is this code trying to make sure that we don't accidentally swallow a ctrl+c?

Reply:
Yeah, the overall goal is to make sure we don't accidentally swallow ctrl+c or other deliberate thread-killing async exceptions, like the ones I believe the `async` package uses for killing threads.